File: LogicalThinkingPatterns/HalfDiamondPattern/HalfDiamond.ts
class HalfDiamond {
  printPattern(n: number): void {
    for (let i = 0; i < n; i++) {
      let pattern: string = "";
      for (let j = 0; j <= i; j++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
    for (let i = n - 1; i > 0; i--) {
      let pattern: string = "";
      for (let j = 0; j < i; j++) {
        pattern += "* ";
      }
      console.log(pattern);
    }
  }
}

const halfDiamond = new HalfDiamond();
halfDiamond.printPattern(5);

File: LogicalThinkingPatterns/HalfDiamondPattern/HalfDiamond.py
class HalfDiamond:
    def printPattern(self, n):
        for i in range(0, n):
            pattern = ""
            for j in range(0, i+1):
                pattern += "* "
            print(pattern)

        for i in range(n-1, 0, -1):
            pattern = ""
            for j in range(0, i):
                pattern += "* "
            print(pattern)
    
    def run(self):
        n = int(input("Enter the number of rows: "))
        self.printPattern(n)

halfDiamond = HalfDiamond()
halfDiamond.run()